crates/bevy_ecs/src/change_detection.rs

```diff
@@ -542,6 +542,45 @@ impl<'a, T: ?Sized> Ref<'a, T> {
     pub fn into_inner(self) -> &'a T {
         self.value
     }
+    /// Map `Ref` to a different type using `f`.
+    ///
+    /// Note that unlike [`Map::map_unchanged`] this doesn't do anything
+    /// special. It's just a more convinient way of constructing a new `Ref`
+    /// from an existing one.
+    pub fn map<U>(self, f: impl FnOnce(&T) -> &U) -> Ref<'a, U> {
+        Ref {
+            value: f(self.value),
+            ticks: self.ticks,
+        }
+    }
+    /// Create a new `Ref` using provided values.
+    ///
+    /// Note that most of the time, you should get `Ref`s as return value
+    /// from engine-internal code.
```

Comment:
Nit: I think this should include a reason why users should think twice before using this everywhere. Would something like this work?
```suggestion
    /// Note that most of the time, you should get `Ref`s as a return value
    /// from engine-internal code in order to prevent accidental misuse.
```

Reply:
I left out this bit because I thought it was harder to misuse than `Mut`, but all things considered it's not a good reason to not include a bit of warning language. Was just a bit lazy. I'll cook up something.

Reply:
I opted to use "This is an advanced features". I'd rather keep the doc short, since, well, the function doesn't do much, and adding on more words is more likely to confuse the reader.